RobotCode/snobot2018/src/org/snobot/elevator/IElevator.java

```diff
@@ -24,8 +24,23 @@
      */
     void setMotorSpeed(double aSpeed);
 
-    void goUp();
+    /**
+     * This sets the height of the elevator.
+     * 
+     * @param aHeight
+     *            is the height that we want to go to in inches.
+     * 
+     */
+    void setHeight(double aHeight);
 
```

Comment:
I think there should be two `setHeight` functions.  One for a raw height, one for an enumerated height.

See https://github.com/ArcticWarriors/snobot-2016/blob/f8d6802f1154d7f32e41de981f1db08d26f11f00/RobotCode/snobot2016/src/com/snobot2016/scaling/IScaling.java#L21 for an example

```
// Pseudo code
enum PresetHeights
{
    Switch(Properties.whatever),
    ScaleLow(Properties.whatever),
    ScaleNeutral(Properties.whatever),
    ScaleHeigh(Properties.whatever);

   double height;
   
    PresetHeights(double height) { height = height; }
 
}

......

BetterElevatorAutoCommand c = new BetterAutoCommand(PresetHeigts.Switch);

if(oi.isSwitchPressed())
{
    elevator.setHeight(PresetHeights.Switch.height); // or pass in the enum and have the class pull the height out
}

Reply:
If this is required we should make another story for it.  I would rather get the claw and climber (if we have one) done before this.  The autonomous modes already can go to different heights.